pkgs/development/python-modules/pycparser/default.nix

```diff
@@ -17,11 +18,22 @@ buildPythonPackage rec {
     hash = "sha256-SRyL6cBA9TkPW/RKWwd1K9B/Vu35kjgbBccBQ57sEPY=";
   };
 
+  postPatch = ''
+    substituteInPlace tests/test_util.py examples/using_cpp_libc.py pycparser/__init__.py \
+      --replace-fail "'cpp'" "'${lib.getBin gcc}/bin/cpp'"
+    substituteInPlace tests/test_util.py examples/using_gcc_E_libc.py \
+      --replace-warn "'gcc'" "'${lib.getBin gcc}/bin/gcc'"
+  '';
+
   build-system = [ setuptools ];
 
-  nativeCheckInputs = [ unittestCheckHook ];
   disabled = pythonOlder "3.8";
 
+  nativeCheckInputs = [
+    unittestCheckHook
+    gcc
```

Comment:
I doubt gcc is correct on plaforms that use clang. Also providing gcc is rarely correct, it should probably be
`pkgsBuildBuild.stdenv.cc` which provides you with both the compiler and linker probably. This way when you cross compile pycparser to musl (pkgsStatic for example) you don't need to compile pkgsStatic.gcc, using the build platform is enough.

This way:
```diff
diff --git a/pkgs/development/python-modules/pycparser/default.nix b/pkgs/development/python-modules/pycparser/default.nix
index 33a8b12fc707..d18091667ce8 100644
--- a/pkgs/development/python-modules/pycparser/default.nix
+++ b/pkgs/development/python-modules/pycparser/default.nix
@@ -5,7 +5,7 @@
   setuptools,
   unittestCheckHook,
   pythonOlder,
-  gcc,
+  pkgsBuildBuild,
 }:
 
 buildPythonPackage rec {
@@ -20,9 +20,9 @@ buildPythonPackage rec {
 
   postPatch = ''
     substituteInPlace tests/test_util.py examples/using_cpp_libc.py pycparser/__init__.py \
-      --replace-fail "'cpp'" "'${lib.getBin gcc}/bin/cpp'"
+      --replace-fail "'cpp'" "'${pkgsBuildBuild.stdenv.cc.targetPrefix}cpp'"
     substituteInPlace tests/test_util.py examples/using_gcc_E_libc.py \
-      --replace-warn "'gcc'" "'${lib.getBin gcc}/bin/gcc'"
+      --replace-warn "'gcc'" "'${pkgsBuildBuild.stdenv.cc.targetPrefix}cc'"
   '';
 
   build-system = [ setuptools ];
@@ -31,7 +31,7 @@ buildPythonPackage rec {
 
   nativeCheckInputs = [
     unittestCheckHook
-    gcc
+    pkgsBuildBuild.stdenv.cc
   ];
 
   unittestFlagsArray = [
```

Reply:
You are right, indeed.